mods/d2k/maps/harkonnen-07/harkonnen07.lua

```diff
@@ -0,0 +1,238 @@
+		Media.DisplayMessage("The Atreides have been annihilated!", "Mentat")
+		player.MarkCompletedObjective(KillAtreides)
+	end
+
+	if corrino.HasNoRequiredUnits() and not player.IsObjectiveCompleted(KillCorrino) then
+		Media.DisplayMessage("The Emperor have been annihilated!", "Mentat")
+		player.MarkCompletedObjective(KillCorrino)
+	end
+
+	if AConYard2.Owner == player and not player.IsObjectiveCompleted(CaptureAtreidesConYard) then
+		player.MarkCompletedObjective(CaptureAtreidesConYard)
+	end
+
+	if (HEngineer.IsDead or AConYard2.IsDead) and not player.IsObjectiveCompleted(CaptureAtreidesConYard) then
+		player.MarkFailedObjective(CaptureAtreidesConYard)
+	end
+
+	if (AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main) and not HiTechIsDead then
+		Media.DisplayMessage("High Tech Factory neutralized! Atreides cut off from Imperial reinforcement!", "Mentat")
+		HiTechIsDead = true
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_main] then
+		local units = ordos_main.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_main] = false
+			ProtectHarvester(units[1], ordos_main, AttackGroupSize[Difficulty])
+		end
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_small] then
+		local units = ordos_small.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_small] = false
+			ProtectHarvester(units[1], ordos_small, AttackGroupSize[Difficulty])
+		end
+	end
+end
+
+WorldLoaded = function()
+	atreides_main = Player.GetPlayer("Atreides Main Base")
+	atreides_small = Player.GetPlayer("Atreides Small Base")
+	corrino = Player.GetPlayer("Corrino")
+	player = Player.GetPlayer("Harkonnen")
+
+	InitObjectives(player)
+	CaptureAtreidesConYard = player.AddPrimaryObjective("Capture the Atreides Construction Yard.")
+	KillAtreides = player.AddPrimaryObjective("Destroy the Atreides.")
+	KillCorrino = player.AddPrimaryObjective("Destroy the Corrino.")
+	KillHarkonnen1 = atreides_main.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen2 = atreides_small.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen3 = corrino.AddPrimaryObjective("Kill all Harkonnen units.")
+
+	Camera.Position = HEngineer.CenterPosition
+	AtreidesAttackLocation = AConYard2.Location
+
+	Trigger.AfterDelay(DateTime.Minutes(5), function()
+		SendAirStrike()
+	end)
+
+	Trigger.OnAllKilledOrCaptured(AtreidesMainBase, function()
+		Utils.Do(atreides_main.GetGroundAttackers(), IdleHunt)
+	end)
+
+	Trigger.OnAllKilledOrCaptured(AtreidesSmallBase, function()
+		Utils.Do(atreides_small.GetGroundAttackers(), IdleHunt)
+	end)
+
+	Trigger.OnAllKilledOrCaptured(CorrinoBase, function()
+		Utils.Do(corrino.GetGroundAttackers(), IdleHunt)
+	end)
+
+	local atreidesWaveCondition = function() return player.IsObjectiveCompleted(KillAtreides) or AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main end
```

Comment:
I noticed this while playing, killing/capturing the tech factory makes the Atreides stop attacking, too. Is this intentional? It's weird that suddenly all attacks stop completely, so I doubt it is.

Reply:
This should only stop the Sonic Tank team from coming. I'm not sure if this is intented or not, i'll check the original mission.

Reply:
Looks like intented. Actually this checked if Emparor is dead in original, but that doesn't make much sense.

![resim](https://user-images.githubusercontent.com/7933210/34090536-cef2ac56-e3c8-11e7-82bc-b3f3aa56322d.png)